src/qtism/runtime/pci/json/Marshaller.php

```diff
@@ -370,10 +371,19 @@ protected function marshallDuration(QtiDuration $duration)
      */
     protected function marshallFile(QtiFile $file)
     {
-        $data = ['base' => ['file' => ['mime' => $file->getMimeType(), 'data' => base64_encode($file->getData())]]];
+        $fileKey = $file instanceof FileHash ? FileHash::FILE_HASH_KEY : 'file';
+
+        $data = [
+            'base' => [
+                $fileKey => [
+                    'mime' => $file->getMimeType(), 
+                    'data' => base64_encode($file->getData())
```

Comment:
It would mean that we need to send base64 decoded file hashes?

Reply:
I'm just keeping the same way as with full file content, as this can be binary, it has to be base64_encoded for inclusion in the json payload. This way, the only different thing is the size of what we store but we store it the same way. So it's more consistent that the calling code provides the hash the same way the full file content is provided. WDYT?

Reply:
Sorry, my mistake. I didn't do the same on both file and filehash in the Unmarshaller but did in the Marshaller... Fixing this.

Reply:
Ok, finally, we're doing so separate things on both File and FIleHash, that we removed the base64 encoding for the hash.